tests/course/test_course_pages.py

```diff
@@ -103,3 +105,30 @@ def test_show_course_page(self, authorized_client, persist_course):
         assert response.status_code == 200
         course_in_page = response.context['course']
         assert persist_course.course_id == course_in_page.course_id
+
+    def test_teacher_accepts_request_to_join_course(self, authorized_client_teacher, student_course0):
+        student_course_id = student_course0.student_course_id
+        teacher_course_id = student_course0.teacher_course_id.course_id
+        assert student_course0.status.value == 'Pending'
+        course_creation_response = authorized_client_teacher.post("/course/" + str(teacher_course_id) + '/' +
+                                                                  str(student_course_id),
+                                                                  {'Accept': 'Accept'}, follow=True)
+        updated_student_course = StudentCourse.objects.filter(student_course_id=student_course_id).first()
+        assert course_creation_response.status_code == 200
+        assert updated_student_course.status == 'Confirmed'
+        expected_template_name = 'course_page.html'
+        page_templates = course_creation_response.templates[0].name
+        assert expected_template_name in page_templates
+
+    def test_teacher_rejects_request_to_join_course(self, authorized_client_teacher, student_course0):
+        student_course_id = student_course0.student_course_id
+        teacher_course_id = student_course0.teacher_course_id.course_id
+        assert student_course0.status.value == 'Pending'
+        course_creation_response = authorized_client_teacher.post("/course/" + str(teacher_course_id) + '/' +
+                                                                  str(student_course_id),
+                                                                  {'Delete': 'Delete'}, follow=True)
+        assert course_creation_response.status_code == 200
+        assert not StudentCourse.objects.filter(student_course_id=student_course_id).exists()
```

Comment:
what about the student_course0.statue.value? does the request itself exists? is it "Rejected"?

Reply:
Sounds like a bug. how the student can know about his request status? I'd argue it should be changed to "Rejected"...

Reply:
Very good! I left one comment regarding the test, fix that and it is ready IMO